packages/loot-core/src/server/db/index.ts

```diff
@@ -623,14 +618,69 @@ export async function getTransactions(accountId) {
   );
 }
 
-export function insertTransaction(transaction) {
+export async function insertTransaction(transaction) {
+  const lastTransaction = await first(
+    'SELECT sort_order FROM v_transactions ' +
+      'WHERE date = ? AND is_child <> 1 AND parent_id IS NULL ' +
+      'ORDER BY sort_order DESC LIMIT 1',
+    [transaction.date.replace(/-/g, '')], // Remove hyphens
+  );
+
+  // Child transactions have relative sort orders.
+  const isChild = transaction.is_child || transaction.parent_id;
+  const sortIncrement =
+    TRANSACTION_SORT_INCREMENT +
+    (isChild && transaction.sort_order ? transaction.sort_order : 0);
+
+  transaction = {
+    ...transaction,
+    sort_order:
+      (lastTransaction ? lastTransaction.sort_order : 0) + sortIncrement,
+  };
+
   return insertWithSchema('transactions', transaction);
 }
 
 export function updateTransaction(transaction) {
   return updateWithSchema('transactions', transaction);
 }
 
+export async function moveTransaction(id, accountId, targetId) {
+  const transactions = await all(
+    'SELECT vt.id, vt.sort_order FROM v_transactions vt WHERE vt.account = ? ' +
+      'AND vt.date = (SELECT vt2.date FROM v_transactions vt2 WHERE vt2.id = ? LIMIT 1) ORDER BY sort_order',
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏if you use \` symbol - you can write multi line strings.